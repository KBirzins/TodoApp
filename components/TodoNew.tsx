import React, {useState} from 'react';
import styled from 'styled-components/native';
import {TouchableWithoutFeedback} from 'react-native';
import {useDispatch} from 'react-redux';

import Icon from './common/Icon';

const StyledView = styled.View`
  flex-direction: row;
  margin-horizontal: 30px;
  margin-vertical: 5px;
  justify-content: flex-start;
  align-items: center;
`;

const StyledText = styled.Text`
  margin-left: 5px;
  font-size: 14px;
  color: grey;
`;

const StyledInput = styled.TextInput`
  margin-left: 5px;
  font-size: 14px;
`;

const TodoNew = () => {
  const [selected, setSelected] = useState(false);
  const [text, setText] = useState('');
  const dispatch = useDispatch();
  return (
    <TouchableWithoutFeedback
      onPress={() => {
        setSelected(!selected);
      }}>
      <StyledView>
        {!selected ? (
          <>
            <Icon color="grey" name="add" />
            <StyledText>New Todo</StyledText>
          </>
        ) : (
          <>
            <Icon color={text !== '' ? 'black' : 'grey'} name="add" />
            <StyledInput
              onChangeText={text => setText(text)}
              value={text}
              onBlur={() => {
                // Dispatch redux action to add new, reset new todo fields
                if (text !== '') {
                  dispatch({type: 'ADD_TODO', text});
                  setText('');
                }
                setSelected(false);
              }}
              autoFocus
            />
          </>
        )}
      </StyledView>
    </TouchableWithoutFeedback>
  );
};

export default TodoNew;
