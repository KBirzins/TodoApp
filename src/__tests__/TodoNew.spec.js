import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import TodoNew from '../components/TodoNew';
import {ReduxProvider} from '../tests/helpers';

describe('TodoNew', () => {
  it('renders correctly with correct properties', () => {
    const component = renderer.create(
      <ReduxProvider>
        <TodoNew />
      </ReduxProvider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
