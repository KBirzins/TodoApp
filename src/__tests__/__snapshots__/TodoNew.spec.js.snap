// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoNew renders correctly with correct properties 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 30,
        "marginVertical": 5,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "color": "grey",
          "fontSize": 25,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î……
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "grey",
          "fontSize": 14,
          "marginLeft": 5,
        },
      ]
    }
  >
    New Todo
  </Text>
</View>
`;
